### DISCUSSION TITLE
Using Sonatype Nexus with kanidm

### DISCUSSION THREAD

**Discussion by tumbl3w33d** [kanidm/kanidm#2757] [Show and tell]
Disclaimer: This is shameful abuse of this platform for my propaganda. I am sorry, but at the same time I would have been glad to be shown that option when I was trying to find a solution to connect my nexus to my kani via OIDC. 😇 

Nexus is an artifact repository which is used in software development toolchains. It comes with lovely features in the free version, however, OIDC is none of them and even in the pro version you only get SAML. There are a few github repos out there with different approaches to handle that situation but none of them really did all I need, so here is my solution:

https://github.com/tumbl3w33d/nexus-oauth2-proxy-plugin

The feature list in the readme explains all the fancy things it does. I tried to keep it simple by making use of the well known oauth2 proxy for the heavy lifting. So since this works nicely with kani (there is an example in the kani docs), your nexus authentication also will. For the programmatic non-interactive access I've chosen a rather opinionated approach inspired by the other repos which tried to solve it. I think it's good enough but I'm open for further suggestions.

Let me know in the issue tracker when you find problems when using it.

**Comment by yaleman** [comment]
I'll let you spam your fancy thing this one time, because you've contributed a lot 😄
